addons/mail/models/mail_thread.py

```diff
@@ -280,10 +280,30 @@ def create(self, vals_list):
             threads_no_subtype = self.env[self._name]
             for thread in threads:
                 subtype = thread._creation_subtype()
-                if subtype:  # if we have a subtype, post message to notify users from _message_auto_subscribe
-                    thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
-                else:
+                if not subtype:
                     threads_no_subtype += thread
+                    continue
+                # if we have a subtype, post message to notify users from _message_auto_subscribe
+                relations = subtype._get_auto_subscription_subtypes(self._name)[4]
+                body = ''
+                if relations:
+                    get_model = self.env["ir.model"]._get
+                    model_name = get_model(self._name).display_name
+                    parent_models = [
+                        f'"{thread[f_name].display_name}" ({get_model(p_model_name).display_name})'
+                        for p_model_name, f_names in relations.items()
+                        for f_name in f_names
+                        if thread[f_name]
+                    ]
+                    if len(parent_models) == 1:
+                        body = _('A new %(model_name)s has been created in the %(parent_model_name)s you follow.',
+                                 model_name=model_name, parent_model_name=parent_models[0])
+                    elif len(parent_models) > 1:
+                        body = _('A new %(model_name)s has been created in one of the %(parent_model_names)s you follow.',
+                                 model_name=model_name, parent_model_names=', '.join(parent_models))
+                    # To hide the message body in the front-end
+                    body = f'<div summary="o_mail_notification">{body}</div>'
+                thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id, body=body)
```

Comment:
A few years ago, I thought we could have the content of a message linked to a subtype being format-friendly. Ide would be: if the "Task Created" subtype could be "{{ object.name }} task created from {{object.project_id.name }} project"  as name / description (aka inline template, to render when the message is posted), we could tune the creation message per app.

We did not do it at that time because it was considered over engineering. Maybe we could think of that solution again, probably to check with Luc. I think it is better to have tailored messages in some main apps, compared to an improve but still too generic message.

Open for discussions anyway :) .